fuel-tx/src/transaction/types/input.rs

```diff
@@ -63,14 +63,16 @@ impl AsField<Vec<u8>> for Vec<u8> {
     }
 }
 
-#[derive(Debug, Clone, PartialEq, Eq, Hash)]
+#[derive(Debug, Clone, PartialEq, Eq, Hash, strum_macros::EnumCount)]
 #[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
 pub enum Input {
     CoinSigned(CoinSigned),
     CoinPredicate(CoinPredicate),
     Contract(Contract),
-    MessageSigned(MessageSigned),
-    MessagePredicate(MessagePredicate),
+    DepositCoinSigned(DepositCoinSigned),
+    DepositCoinPredicate(DepositCoinPredicate),
+    MetadataSigned(MetadataSigned),
```

Comment:
Why are we changing the name from message to metadata? We don't have anything in the specs referring to metadata inputs so this is confusing.

Reply:
changed to MessageCoin_ and MessageData_